### PR TITLE
refactor: mark deprecated config variables clearer

### PR DESCRIPTION
**Pull request by scop**
Move ones in the main `bash_completion` to the compat snippet.

### CODE REVIEW

File: bash_completion.d/000_bash_completion_compat.bash [hunk 1]
```diff
@@ -178,6 +178,14 @@ _realcommand()
     return $rc
 }
 
+# @deprecated set the `BASH_COMPLETION_*` variables directly instead.
+[[ -v COMP_FILEDIR_FALLBACK ]] &&
+    : "${BASH_COMPLETION_FILEDIR_FALLBACK:=$COMP_FILEDIR_FALLBACK}"
```
Comment (akinomyoga) [inline bash_completion.d/000_bash_completion_compat.bash:183]:
```suggestion
    : "${BASH_COMPLETION_FILEDIR_FALLBACK=$COMP_FILEDIR_FALLBACK}"
```

When there are both settings `BASH_COMPLETION_FILEDIR_FALLBACK=` and `COMP_FILEDIR_FALLBACK=set`, I think `BASH_COMPLETION_FILEDIR_FALLBACK` should have higher precedence.

**Review by akinomyoga — COMMENTED** [review]
Some non-trivial question is whether to support the case those old variables are changed after sourcing `bash_completion`. But anyway we should remove these special treatments for old names from the main part of the script at some point.

In `bash >= 4.3`, we may declare

```bash
declare -gn COMP_FILEDIR_FALLBACK=BASH_COMPLETION_FILEDIR_FALLBACK
```

in `000_bash_completion_compat.bash`.

File: bash_completion.d/000_bash_completion_compat.bash (hunk 1 above)
Comment (scop) [reply bash_completion.d/000_bash_completion_compat.bash]: Right! Addressed in 8e855a2c3057fb745d28e23728bb2d7edd182379.

**Comment by scop** [comment]
Hmm, true that. I'll mark this as a draft and let it cook for a bit.

File: bash_completion.d/000_bash_completion_compat.bash (hunk 1 above)
Comment (akinomyoga) [reply bash_completion.d/000_bash_completion_compat.bash]: Thanks!

**Comment by scop** [comment]
Modified this some, guess it's ready for review if this is something we'd want.

**Review by akinomyoga — APPROVED** [review]
I agree with merging it. Bash 4.2 is anyway old and not used in many environments.

File: bash_completion [hunk 2]
```diff
@@ -2520,7 +2519,7 @@ _known_hosts_real()
 
     # Add results of normal hostname completion, unless
     # `BASH_COMPLETION_KNOWN_HOSTS_WITH_HOSTFILE' is set to an empty value.
-    if [[ ${BASH_COMPLETION_KNOWN_HOSTS_WITH_HOSTFILE-${COMP_KNOWN_HOSTS_WITH_HOSTFILE-1}} ]]; then
+    if [[ ${BASH_COMPLETION_KNOWN_HOSTS_WITH_HOSTFILE-set ]]; then
```
[the remark this answers is not in the document]
Comment (akinomyoga) [reply bash_completion]: Ah, please ignore this comment. This seems to have been *a pending comment* for about half an year.

**Pull request merged**